### PR TITLE
Move boskos deployment under prow, auto-deploy

### PR DESCRIPTION
**Pull request by fejta**
/assign @Katharine @SebastianM @krzyzacy @chases2 
/hold

* Merge the boskos service object into the main boskos file.
* Move the unit test to validate that the config parses into prow/cluster/boskos_test.go
* Replace all Makefile with bazel rules

fixes  https://github.com/kubernetes/test-infra/issues/14267

### CODE REVIEW

**Comment by krzyzacy** [comment]
there are other stuff:

https://github.com/kubernetes/test-infra/blob/master/boskos/janitor/deployment.yaml
https://github.com/kubernetes/test-infra/blob/master/boskos/reaper/deployment.yaml
https://github.com/kubernetes/test-infra/blob/master/boskos/metrics/deployment.yaml

File: prow/cluster/boskos.yaml [hunk 1]
```diff
@@ -11,10 +11,6 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-apiVersion: v1
```
Comment (sebastienvas) [inline prow/cluster/boskos.yaml]: Why ? is this already part of the BUILD_CLUSTER definition ?

Comment (fejta) [reply prow/cluster/boskos.yaml]: This is an invalid yaml file. The following three lines of code overwrite these three.

And yes, a) the test-pods namespace already exists and b) we should not be define it in boskos.yaml

**Comment by fejta** [comment]
Passing tests!

**Comment by fejta** [comment]
Let's merge this tomorrow? 1.16 should be done then.

**Pull request merged**